File: G50/Sebastian_Rojas_Vargas/Diagnostico1/PunterosFutbolJava/src/Main/Metodos.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package Main;

/**
 *
 * @author Sebas
 */
public class Metodos {

    Jugador inicio, ultimo;

    public boolean insertarOrdenado(String nombre, int numeroCamisa) {
        Jugador nuevo = new Jugador(nombre, numeroCamisa);
        if (inicio == null) {
            inicio = ultimo = nuevo;
            return true;
        }
        if ((inicio.numeroCamisa == nuevo.numeroCamisa) | (ultimo.numeroCamisa == nuevo.numeroCamisa)) {
            return false;
        }
        if (nuevo.numeroCamisa < inicio.numeroCamisa) {
            nuevo.sig = inicio;
            inicio.ant = nuevo;
            inicio = nuevo;
            return true;
        }
        if (nuevo.numeroCamisa > ultimo.numeroCamisa) {
            ultimo.sig = nuevo;
            nuevo.ant = ultimo;
            ultimo = nuevo;
            return true;
        }
        Jugador aux = inicio;
        while (aux != null) {
            if (nuevo.numeroCamisa == aux.numeroCamisa) {
                return false;
            }
            if (nuevo.numeroCamisa < aux.numeroCamisa) {
                aux.ant.sig = nuevo;
                nuevo.ant = aux.ant;
                nuevo.sig = aux;
                aux.ant = nuevo;
                return true;
            }
            aux = aux.sig;
        }
        return false;
    }

    public void mostrar() {
        Jugador aux = inicio;
        while (aux != null) {
            System.out.println(aux.nombre + " " + aux.numeroCamisa);
            aux = aux.sig;
        }
    }
}

File: G50/Sebastian_Rojas_Vargas/Diagnostico1/FutboPunterosPython/Metodos.py
from Jugador import *

inicio = None
ultimo = None

def insertarOrdenado(nombre, numeroCamisa):
    nuevo = Jugador()
    nuevo.set__info(nombre, numeroCamisa)
    global inicio
    global ultimo
    if (inicio == None):
        inicio = ultimo = nuevo
        return True
        
    if ((inicio.numeroCamisa == nuevo.numeroCamisa) | (ultimo.numeroCamisa == nuevo.numeroCamisa)):
        return False
        
    if (nuevo.numeroCamisa < inicio.numeroCamisa):
        nuevo.sig = inicio
        inicio.ant = nuevo
        inicio = nuevo
        return True
        
    if (nuevo.numeroCamisa > ultimo.numeroCamisa):
        ultimo.sig = nuevo
        nuevo.ant = ultimo
        ultimo = nuevo
        return True
    
    aux = inicio
    while (aux != None):
        if (nuevo.numeroCamisa == aux.numeroCamisa):
            return False
        
        if (nuevo.numeroCamisa < aux.numeroCamisa):
            aux.ant.sig = nuevo
            nuevo.ant = aux.ant
            nuevo.sig = aux
            aux.ant = nuevo
            return True
            
        aux = aux.sig
        
    return False

def mostrar():
    aux = inicio
    while (aux != None):
        aux.get__info()
        aux = aux.sig










